internal/pkg/levee/levee_ear_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package levee
+
+import (
+	"testing"
+
+	"github.com/google/go-flow-levee/internal/pkg/debug"
+	"golang.org/x/tools/go/analysis/analysistest"
+)
+
+func TestLeveeEAR(t *testing.T) {
+	dataDir := analysistest.TestData()
+	if err := Analyzer.Flags.Set("config", dataDir+"/test-config.yaml"); err != nil {
+		t.Error(err)
+	}
+	if err := Analyzer.Flags.Set("useEAR", "true"); err != nil {
+		t.Error(err)
+	}
+	if *debugging {
+		Analyzer.Requires = append(Analyzer.Requires, debug.Analyzer)
+	}
+
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/arguments")
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/basictypes")
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/binop")
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/booleans")
+	// analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/callorder")  // TODO: flow-sensitive
```

Comment:
If these test differences are irreconcilable, we should partition them out.
 
Part of the appeal of the `...` format is that we were forgetting to add our test targets when we added new test packages.  We should refactor these tests soon so that the shared tests can be `example/tests/shared/...`, and approach-specific tests can be tested at `example/tests/propagation/...` and `example/tests/ear/...`.  That can happen in a future PR, though.

Reply:
Good point. I added a TODO #318 for this.